File: mealroll/tests/run-all.js
const { execSync } = require('child_process');
const path = require('path');
const tests = ['recipes.test.js', 'spin.test.js'];
let allPassed = true;
const results = [];

console.log('MealRoll Test Suite\n');

tests.forEach(test => {
  try {
    const output = execSync(`node "${path.join(__dirname, test)}"`, { encoding: 'utf8', stdio: 'pipe' });
    console.log(output);
    results.push({ test, status: 'PASS' });
  } catch (err) {
    console.log(err.stdout || '');
    console.error(err.stderr || '');
    results.push({ test, status: 'FAIL' });
    allPassed = false;
  }
});

console.log('\n=== Summary ===');
results.forEach(r => console.log(`  ${r.status === 'PASS' ? '\u2705' : '\u274c'} ${r.test}`));
const p = results.filter(r => r.status === 'PASS').length;
console.log(`\n${p}/${results.length} test suites passed`);
process.exit(allPassed ? 0 : 1);

File: nestplan/tests/run_all.py
"""Run all NestPlan tests."""

import subprocess
import sys
import os

test_dir = os.path.dirname(os.path.abspath(__file__))
tests = [
    "test_models.py",
    "test_storage.py",
    "test_budget.py",
]

all_passed = True
results = []

print("NestPlan Test Suite\n")

for test in tests:
    filepath = os.path.join(test_dir, test)
    try:
        result = subprocess.run(
            [sys.executable, filepath],
            capture_output=True, text=True, timeout=30,
        )
        print(result.stdout, end="")
        if result.stderr:
            print(result.stderr, end="")
        if result.returncode == 0:
            results.append((test, "PASS"))
        else:
            results.append((test, "FAIL"))
            all_passed = False
    except Exception as e:
        print(f"  ERROR running {test}: {e}")
        results.append((test, "ERROR"))
        all_passed = False

print("\n=== Summary ===")
for name, status in results:
    icon = "\u2705" if status == "PASS" else "\u274c"
    print(f"  {icon} {name}")

total = len(results)
passing = sum(1 for _, s in results if s == "PASS")
print(f"\n{passing}/{total} test suites passed")

sys.exit(0 if all_passed else 1)
